---
title: "Microbiome and Community analysis"
author: "Kendra Maas"
date: "May 12, 2025"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# UConn MARS Workshop 

Microbes control the world! And with advances in molecular techniques, esp. DNA sequencing, we can explore the microbial side with ever increasing depth and breadth. But the data generated by sequencing the microbial world is bigish and weird. Most bioinformatic tools don't handle the taxonomic markers that we like to sequence very well. There are a few specialized programs for processing phylogenetic marker sequences, my preferred program is mothur <http://www.mothur.org/wiki/MiSeq_SOP>. We will use mothur generated OTU table (species by sample matrix) into R to explore the data and learn how to make publication quality figures. 

Basic Schedule. We will roughly follow this, but the exact amount of time we spend on each topic will be adjusted depending on the class.

In class discussions and commands will be posted in an etherpad <https://etherpad.mit.edu/p/20250512MEworkshop>


Day|Session|Topic
------|-------|--------
Day 1|Morning|[Introduction to communities](https://krmaas.github.io/20250512MEworkshop/intro.communities.html)
.|.|[Overview of wetlab](https://krmaas.github.io/20250512MEworkshop/introWetLab.html) 
.|.|[Potential wetlab errors](https://krmaas.github.io/20250512MEworkshop/wetLabErrors.html)
.|Afternoon| Introduction to Basespace and Command Line 
.|.|Begin mothur processing
.|.|Sequencing quality control
Day 2|Morning|Continue mothur processing
.|.|Chimera checking
.|.|Taxonomic assignment
.|Afternoon|OTU clustering
.|.|Calculate alpha and beta diversity
Day 3|Morning|Reproducible Research
.|.|Introduction to R ggplot
.|Afternoon|Importing data into R
.|.|Import mothur into R
.|.|Subsetting, joining, data management
Day 4|Morning|Ecological statistics
.|Afternoon|Hypothesis testing & Selecting indicator OTUs
